src/CoreMedia/CMCustomBlockAllocator.cs

```diff
@@ -59,7 +62,9 @@ internal struct CMBlockBufferCustomBlockSource {
 		static IntPtr AllocateCallback (IntPtr refCon, nuint sizeInBytes)
 		{
 			var gch = GCHandle.FromIntPtr (refCon);
-			return ((CMCustomBlockAllocator) gch.Target).Allocate (sizeInBytes);
+			if (gch.Target is CMCustomBlockAllocator target)
+				return target.Allocate (sizeInBytes);
+			return IntPtr.Zero;
```

Comment:
Not sure if this is correct

Reply:
Looks good to me, can't be much worse than throwing a InvalidCastException.